Kinvey/Kinvey/User.swift

```diff
@@ -42,15 +49,15 @@ open class User: NSObject, Credential, Mappable {
     
     /// Creates a new `User` taking (optionally) a username and password. If no `username` or `password` was provided, random values will be generated automatically.
     @discardableResult
-    open class func signup(username: String? = nil, password: String? = nil, client: Client = Kinvey.sharedClient, completionHandler: UserHandler? = nil) -> Request {
+    open class func signup<UserType: User>(username: String? = nil, password: String? = nil, user: UserType? = nil, client: Client = Kinvey.sharedClient, completionHandler: ((UserType?, Swift.Error?) -> Void)? = nil) -> Request {
         precondition(client.isInitialized(), "Client is not initialized. Call Kinvey.sharedClient.initialize(...) to initialize the client before attempting to sign up.")
 
-        let request = client.networkRequestFactory.buildUserSignUp(username: username, password: password)
-        Promise<User> { fulfill, reject in
+        let request = client.networkRequestFactory.buildUserSignUp(username: username, password: password, user: user)
+        Promise<UserType> { fulfill, reject in
             request.execute() { (data, response, error) in
                 if let response = response , response.isOK {
                     client.activeUser = client.responseParser.parseUser(data)
-                    fulfill(client.activeUser!)
+                    fulfill(client.activeUser as! UserType)
```

Comment:
This would require that `client.activeUser` returns a `UserType` instance. Is that true today? Last I checked, that wasn't the case.